File: study/lua/src/app/scene/view/cakeActivity/CakePlayerIcon.lua
--
-- Author: Liangxu
-- Date: 2019-4-30
-- 蛋糕活动军团Icon

local CakePlayerIcon = class("CakeGuildIcon")
local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local CakeActivityDataHelper = require("app.utils.data.CakeActivityDataHelper")

local RES_INFO = {
	[1] = {
		rankRes = "img_ranking_01", 
		textRes = "txt_ranking01",
		color = cc.c3b(0xff, 0xd8, 0x00),
		scale = 0.6,
	},
	[2] = {
		rankRes = "img_ranking_02", 
		textRes = "txt_ranking02",
		color = cc.c3b(0xb6, 0xc8, 0xeb),
		scale = 0.5,
	},
	[3] = {
		rankRes = "img_ranking_03", 
		textRes = "txt_ranking03",
		color = cc.c3b(0xed, 0x8f, 0x50),
		scale = 0.5,
	},
}

function CakePlayerIcon:ctor(target, rank)
	self._target = target

	self._nodeIcon = ccui.Helper:seekNodeByName(self._target, "NodeIcon")
	cc.bind(self._nodeIcon, "CommonHeroIcon")
	self._imageRank = ccui.Helper:seekNodeByName(self._target, "ImageRank")
	self._imageText = ccui.Helper:seekNodeByName(self._target, "ImageText")
	self._textScore = ccui.Helper:seekNodeByName(self._target, "TextScore")
	self._textName = ccui.Helper:seekNodeByName(self._target, "TextName")

	self:_initUI(rank)
end

function CakePlayerIcon:_initUI(rank)
	local info = RES_INFO[rank]
	if info then
		self._imageText:loadTexture(Path.getAnniversaryImg(info.textRes))
		self._textName:setColor(info.color)
		self._imageRank:loadTexture(Path.getTextSignet(info.rankRes))
		self._nodeIcon:setScale(info.scale)
	end
end

function CakePlayerIcon:updateUI(data)
	if data then
		local coverId = data:getCovertIdAndLimitLevel()
		self._nodeIcon:updateUI(coverId)
		self._nodeIcon:updateHeadFrame(data:getHead_frame_id())
		self._nodeIcon:showHeroUnknow(false)
		self._textScore:setString(data:getPoint())
		local strName = data:getName()
		if data:getServer_name() ~= "" then
			local serverName = require("app.utils.TextHelper").cutText(data:getServer_name())
			strName = serverName.."\n"..strName
		end
		self._textName:setString(strName)
	else
		self._nodeIcon:showHeroUnknow(true)
		self._textScore:setString("")
		self._textName:setString(Lang.get("cake_activity_icon_empty"))
	end
end

return CakePlayerIcon

File: study/ts/script/scene/view/cakeActivity/CakePlayerIcon.ts
import CommonHeroIcon from "../../../ui/component/CommonHeroIcon";
import { Path } from "../../../utils/Path";
import UIHelper from "../../../utils/UIHelper";
import { CakeActivityDataHelper } from "../../../utils/data/CakeActivityDataHelper";
import { Lang } from "../../../lang/Lang";
import { TextHelper } from "../../../utils/TextHelper";

const { ccclass, property } = cc._decorator;

@ccclass
export default class CakePlayerIcon extends cc.Component {
    public static readonly RES_INFO = {
        [1]: {
            rankRes: 'img_ranking_01',
            textRes: 'txt_ranking01',
            color: new cc.Color(255, 216, 0),
            scale: 0.6
        },
        [2]: {
            rankRes: 'img_ranking_02',
            textRes: 'txt_ranking02',
            color: new cc.Color(182, 200, 235),
            scale: 0.5
        },
        [3]: {
            rankRes: 'img_ranking_03',
            textRes: 'txt_ranking03',
            color: new cc.Color(237, 143, 80),
            scale: 0.5
        }
    }
    _target: cc.Node;

    @property({
        type: CommonHeroIcon,
        visible: true
    })
    _nodeIcon: CommonHeroIcon = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageRank: cc.Sprite = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageText: cc.Sprite = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textScore: cc.Label = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textName: cc.Label = null;

    ctor(rank) {
        this._initUI(rank);
    }
    _initUI(rank) {
        var info = CakePlayerIcon.RES_INFO[rank];
        if (info) {
            UIHelper.loadTexture(this._imageRank, Path.getTextSignet(info.rankRes))
            UIHelper.loadTexture(this._imageText, Path.getAnniversaryImg(info.textRes))

            this._textName.node.color = (info.color);
            this._nodeIcon.node.setScale(info.scale);
        }
    }
    updateUI(data) {
        if (data) {
            var coverId = data.getCovertIdAndLimitLevel()[0];
            this._nodeIcon.updateUI(coverId);
            this._nodeIcon.updateHeadFrame(data.getHead_frame_id());
            this._nodeIcon.showHeroUnknow(false);
            this._textScore.string = (data.getPoint());
            var strName = data.getName();
            if (data.getServer_name() != '') {
                var serverName = TextHelper.cutText(data.getServer_name());
                strName = serverName + ('\n' + strName);
            }
            this._textName.string = (strName);
        } else {
            this._nodeIcon.showHeroUnknow(true);
            this._textScore.string = ('');
            this._textName.string = (Lang.get('cake_activity_icon_empty'));
        }
    }

}